subprojects/model-core/src/test/groovy/org/gradle/internal/reflect/validation/ValidationMessageCheckerTest.groovy

```diff
@@ -354,7 +354,7 @@ Reason: A property of type 'FileCollection' annotated with @Input cannot determi
 
 Possible solutions:
   1. Annotate with @InputFile for regular files.
-  2. Annotate with @InputDirectory for directories.
+  2. Annotate with @InputFiles for collections of files.
```

Comment:
Why remove the `@InputDirectory` solution here? Then you'd need to remove the `@InputFile` solution as well.

Reply:
I was hoping to offer finer-grained errors when `@Input` was misapplied to a file vs. a directory.  I split the checks in `InputPropertyAnnotationHandler` into 2 separate validations for each of these.

I know `Path` could represent a directory, but I figure it's typically a file and putting `@Input` on a FileCollection seemed like a more common error to me.  Maybe this case does need to retain an additional possible solution.  Maybe instead of this: "Annotate with `@InputDirectory` for `Path`s representing directories" since other types of directories will be checked separately now?